src/wallet/wallet.cpp

```diff
@@ -159,9 +162,13 @@ bool CWallet::LoadKeyMetadata(const CPubKey &pubkey, const CKeyMetadata &meta)
     return true;
 }
 
-bool CWallet::LoadCryptedKey(const CPubKey &vchPubKey, const std::vector<unsigned char> &vchCryptedSecret)
+bool CWallet::LoadCryptedKey(const CPubKey &vchPubKey, const std::vector<unsigned char> &vchCryptedSecret, const std::vector<unsigned char> &vchHash)
 {
-    return CCryptoKeyStore::AddCryptedKey(vchPubKey, vchCryptedSecret);
+    std::vector<unsigned char> vchCheckHash;
+    CCryptoKeyStore::CalculateCryptedHash(vchPubKey, vchCryptedSecret, vchCheckHash);
+    if (!std::equal(vchHash.begin(), vchHash.end(), vchCheckHash.begin()))
```

Comment:
not a strong one, i used uint256 for the hash over unencrypted keys

i'll switch to uint256